### PR TITLE
Added CI for documentation

### LINKED ISSUE #1099: Add Sphinx Build to CI

### PR DESCRIPTION
**Pull request by rainman110**
## Description

This adds a Github Action to create the developer documentation.
Currently, the documentation is stored as a build artifact only.

Later, when the documentation has been finished, we should deploy
it to a website, e.g. github pages.

## How Has This Been Tested?

See the CI results....

## Checklist:

- [ ] A test for the new functionality was added (if applicable).
- [ ] All tests run without failure.
- [ ] The changelog has been extended, if this MR contains important changes from the users's point of view.
- [ ] The new code complies with the GTlab's style guide.
- [ ] New interface methods / functions are exported via `EXPORT`. Non-interface functions are NOT exported.
- [ ] The number of code quality warnings is not increasing.

Closes #1099

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: CI
+
+on:
+  push:
+    branches: [ "master" ]
```
Comment (jensschmeink) [inline .github/workflows/ci.yml:5]: A question for the development of the content of the documentation: Currently the documentaion will only be build for the master. How would it look like to build it always (while the development). Could I just remove tht "on:" part temporary?

**Review by jensschmeink — APPROVED** [review]
Overall I like this. Just have one question (just fo furhter development)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rainman110) [reply .github/workflows/ci.yml:5]: The documentation is build on master and whenever a pull request want to merge in master. I think this is sufficient.

If you want to build always, simply remove the `on` block.

Comment (rainman110) [reply .github/workflows/ci.yml:5]: As you can see, the CI was executed on this branch as well, since it was part of a pull request.

**Pull request merged**